src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/MacRazorProjectHostBase.cs

```diff
@@ -236,8 +236,10 @@ protected void RemoveDocument(HostProject hostProject, string filePath)
                 return false;
             }
 
-            var basePath = new DirectoryInfo(baseIntermediateOutputPathValue).Parent;
-            var joinedPath = Path.Combine(basePath.FullName, intermediateOutputPathValue);
+            var normalizedBaseIntermediateOutputPath = ToPlatformFilePath(baseIntermediateOutputPathValue);
```

Comment:
This whole call path resolves the `IntermediateOutputPath`, aka the `obj\Debug` directory which is where we typically store our project.razor.json state. Because we couldn't resolve where that was we'd bail out early on even figuring out where to publish a project.razor.json. Therefore VSMac wouldn't see it 😄